test/integration/xds_config_tracker_integration_test.cc

```diff
@@ -178,6 +214,34 @@ TEST_P(XdsConfigTrackerIntegrationTest, XdsConfigTrackerSuccessCount) {
 
   // onConfigAccepted is called when all the resources are accepted.
   test_server_->waitForCounterEq("test_xds_tracker.on_config_accepted", 1);
+  test_server_->waitForCounterEq("test_xds_tracker.on_config_metadata_read", 0);
+  test_server_->waitForCounterEq("test_xds_tracker.on_config_metadata_packed", 0);
+  EXPECT_EQ(1, test_server_->counter("test_xds_tracker.on_config_accepted")->value());
+}
+
+// This is to test the Resource wrapper usage with metadata field.
+TEST_P(XdsConfigTrackerIntegrationTest, XdsConfigTrackerSuccessCountWithWrapper) {
+  TestXdsConfigTrackerFactory factory;
+  Registry::InjectFactory<Config::XdsConfigTrackerFactory> registered(factory);
+
+  initialize();
+  EXPECT_TRUE(compareDiscoveryRequest(Config::TypeUrl::get().Cluster, "", {}, {}, {}, true));
+
+  // Add a typed metadata to the Resource wrapper.
+  test::envoy::config::xds::TestTrackerMetadata test_metadata;
+  ProtobufWkt::Any packed_value;
+  packed_value.PackFrom(test_metadata);
+  sendDiscoveryResponse<envoy::config::cluster::v3::Cluster>(
+      Config::TypeUrl::get().Cluster, {cluster1_, cluster2_}, {cluster1_, cluster2_}, {}, "1",
+      {{kTestKey, packed_value}});
+
+  // 3 because the statically specified CDS server itself counts as a cluster.
+  test_server_->waitForGaugeGe("cluster_manager.active_clusters", 3);
+
+  // onConfigAccepted is called when all the resources are accepted.
+  test_server_->waitForCounterEq("test_xds_tracker.on_config_accepted", 1);
+  test_server_->waitForCounterEq("test_xds_tracker.on_config_metadata_read", 2);
+  test_server_->waitForCounterEq("test_xds_tracker.on_config_metadata_packed", 2);
```

Comment:
nit: seems that "packed" and "read" always have the same value in the tests. If there's no added benefit, should probably use a single one of these.

Reply:
Thanks for the review! Done.